client/state/data-layer/wpcom/sites/comments/change-status/index.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+import { get, truncate } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { stripHTML, decodeEntities } from 'lib/formatting';
+import { COMMENTS_CHANGE_STATUS } from 'state/action-types';
+import { mergeHandlers } from 'state/action-watchers/utils';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { local } from 'state/data-layer/utils';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { errorNotice } from 'state/notices/actions';
+import { getSiteComment } from 'state/selectors';
+
+const changeCommentStatus = ( { dispatch, getState }, action ) => {
+	const { siteId, commentId, status } = action;
+	const previousStatus = get( getSiteComment( getState(), action.siteId, action.commentId ), 'status' );
+
+	dispatch(
+		http(
+			{
+				method: 'POST',
+				path: `/sites/${ siteId }/comments/${ commentId }`,
+				apiVersion: '1.1',
+				body: {
+					status,
+				},
+			},
+			{
+				...action,
+				previousStatus,
+			}
+		)
+	);
+};
+
+const verifyCommentStatus = ( { dispatch, getState }, action, next, data ) => {
+	const statusFromState = get( getSiteComment( getState(), action.siteId, action.commentId ), 'status' );
+	const statusFromServer = get( data, 'status' );
+
+	if ( statusFromServer && statusFromServer !== statusFromState ) {
+		dispatch(
+			local( {
+				...action,
+				status: get( data, 'status' ),
+			} )
+		);
+	}
+};
+
+const getErrorMessage = ( status, content ) => {
+	const commentContent = content && truncate( decodeEntities( stripHTML( content ) ), { length: 30, omission: '…' } );
+
+	switch ( status ) {
+		case 'approved':
+			return commentContent
+				? translate( 'Could not approve the comment "%(commentContent)s".', { args: { commentContent } } )
+				: translate( 'Could not approve the comment.' );
+		case 'unapproved':
+			return commentContent
+				? translate( 'Could not unapprove the comment "%(commentContent)s".', { args: { commentContent } } )
+				: translate( 'Could not unapprove the comment.' );
```

Comment:
Hi! I've found a possible matching string that has already been translated **98** times:
`translate( 'Unapprove this comment' )` *ES Score: 12*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/data-layer/change-comment-status)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).